app/models/entitlement.rb

```diff
@@ -17,8 +17,15 @@ def self.valid_filters?(filters_hash)
     filters_hash["managed"].present? || filters_hash["belongsto"].present?
   end
 
+  def self.update_managed_filters(old_tag, new_tag)
+    where.not(:filters => nil).find_each do |entitlement|
+      entitlement.update_managed_filters(old_tag, new_tag)
+      entitlement.save if entitlement.filters_changed?
+    end
+  end
+
   def self.remove_tag_from_all_managed_filters(tag)
-    find_each do |entitlement|
```

Comment:
This is optimization 1, somewhat related to the change in this PR but can be extracted if distracting.  Basically, I realized we were checking `nil` filters in ruby when we can avoid that completely by removing `NULL` filters in SQL and so only process ones that could possibly have managed filters.